tests/unit/services/test_package.py

```diff
@@ -123,9 +123,11 @@ def test_write_metadata(
         build_plan=default_build_plan,
     )
 
-    package_service.write_metadata(new_dir)
+    prime_dir = new_dir / "prime"
+
```

Comment:
```suggestion
    prime_dir = new_dir / "prime"
    meta_dir = prime_dir / "meta"

```

So it looks like https://github.com/canonical/snapcraft/pull/4661/files#diff-30d06cd9fb0ded435b9c8e2324ae541a4b0c1100e9802c398d57860f4709f048L205